cnf-tests/testsuites/e2esuite/dpdk/dpdk.go

```diff
@@ -329,6 +341,69 @@ var _ = Describe("dpdk", func() {
 		})
 	})
 
+	Context("dpdk application on different vendors", func() {
+		var nodeNames []string
+		var sriovInfos *sriovcluster.EnabledNodes
+		var err error
+		var out string
+
+		execute.BeforeAll(func() {
+			sriovInfos, err = sriovcluster.DiscoverSriov(sriovclient, namespaces.SRIOVOperator)
+			Expect(err).ToNot(HaveOccurred())
+
+			Expect(sriovInfos).ToNot(BeNil())
+
+			nodeNames, err = nodes.MatchingOptionalSelectorByName(sriovInfos.Nodes)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(len(nodeNames)).To(BeNumerically(">", 0))
+		})
+
+		BeforeEach(func() {
+			if discovery.Enabled() {
+				Skip("Split VF test disabled for discovery mode")
+			}
+
+			CleanSriov()
+		})
+
+
+		DescribeTable("Test connectivity using the requested nic", func(vendorID, deviceID string) {
+			By("searching for the requested network card")
+			node, sriovDevice,exist := findSriovDeviceForDPDK(sriovInfos,nodeNames,vendorID, deviceID)
+			if ! exist {
+				Skip(fmt.Sprintf("skip nic validate as wasn't able to find a nic with vendorID %s and deviceID %s", vendorID, deviceID))
+			}
+
+			By("creating a network policy")
+			createPoliciesDPDKOnly(sriovDevice, node, dpdkResourceName)
+
+			By("creating a network")
+			CreateSriovNetwork(sriovDevice, "test-dpdk-network", dpdkResourceName)
+
+			By("creating a pod")
+			txOnlydpdkWorkloadPod , err := createDPDKWorkload(nodeSelector,
+				strings.ToUpper(dpdkResourceName),
+				fmt.Sprintf(CLIENT_TESTPMD_COMMAND, strings.ToUpper(dpdkResourceName)),
+				5,
+				false)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Parsing output from the DPDK application")
+			Eventually(func() string {
+				out, err = pods.GetLog(txOnlydpdkWorkloadPod)
+				Expect(err).ToNot(HaveOccurred())
+				return out
+			}, 8*time.Minute, 1*time.Second).Should(ContainSubstring(LOG_ENTRY),
```

Comment:
Is this right? The test passes if the string is found in the logs?

Reply:
then we ahve the `checkTxOnly(out)` in line 398